File: lib/domain/repositories/library_repository.dart
import 'package:libcheck/domain/models/book_availability.dart';
import 'package:libcheck/domain/models/library.dart';

abstract class LibraryRepository {
  Future<List<Library>> getLibraries({
    required String pref,
    String? city,
  });

  Future<List<BookAvailability>> checkBookAvailability({
    required List<String> isbn,
    required List<String> systemIds,
  });
}

File: src/domain/repositories/libraryRepository.ts
import { BookAvailability } from '@/domain/models/bookAvailability';
import { Library } from '@/domain/models/library';

export interface LibraryRepository {
  getLibraries(args: { pref: string; city?: string }): Promise<Library[]>;
  checkBookAvailability(args: {
    isbn: string[];
    systemIds: string[];
  }): Promise<BookAvailability[]>;
}
